src/admin.rs

```diff
@@ -65,20 +66,33 @@ pub fn load_object_store_from_env(
 /// | AWS_SECRET_ACCESS_KEY | The access key secret for the above ID | Yes |
 /// | AWS_BUCKET | The bucket to use within S3 | Yes |
 /// | AWS_REGION | The AWS region to use | Yes |
+/// | AWS_ENDPOINT | The endpoint to use for S3 (disables https) | No |
+/// | AWS_DYNAMODB_TABLE | The DynamoDB table to use for locking | No |
 #[cfg(feature = "aws")]
 pub fn load_aws() -> Result<Arc<dyn ObjectStore>, Box<dyn Error>> {
+    use object_store::aws::{DynamoCommit, S3ConditionalPut};
+
     let key = env::var("AWS_ACCESS_KEY_ID").expect("AWS_ACCESS_KEY_ID must be set");
     let secret =
         env::var("AWS_SECRET_ACCESS_KEY").expect("Expected AWS_SECRET_ACCESS_KEY must be set");
     let bucket = env::var("AWS_BUCKET").expect("AWS_BUCKET must be set");
     let region = env::var("AWS_REGION").expect("AWS_REGION must be set");
-
-    Ok(Arc::new(
-        object_store::aws::AmazonS3Builder::new()
-            .with_access_key_id(key)
-            .with_secret_access_key(secret)
-            .with_bucket_name(bucket)
-            .with_region(region)
-            .build()?,
-    ) as Arc<dyn ObjectStore>)
+    let endpoint = env::var("AWS_ENDPOINT").ok();
+    let dynamodb_table = env::var("AWS_DYNAMODB_TABLE").ok();
+    let builder = object_store::aws::AmazonS3Builder::new()
+        .with_access_key_id(key)
+        .with_secret_access_key(secret)
+        .with_bucket_name(bucket)
+        .with_region(region);
+    let builder = if let Some(endpoint) = endpoint {
+        builder.with_allow_http(true).with_endpoint(endpoint)
+    } else {
+        builder
+    };
+    let builder = if let Some(dynamodb_table) = dynamodb_table {
```

Comment:
should we just error out if this is not set? I think the db will just fail anyway

Reply:
Good call. It's a little tricky since some of the commands in the admin CLI can work without DynamoDB. But for bencher, it's required. I've opted to panic if it's not set for now. It'll all go away once `object_store` adds CAS for S3, anyway.

Reply:
https://github.com/slatedb/slatedb/pull/247/commits/51bc91eae993298e6d6624877908f38ce145981f